### ISSUE TITLE
ChannelCachingConnectionProxy close

### ISSUE THREAD

**Issue by wang007** [spring-projects/spring-amqp#1186]
why CachingConnectionFactory  allocatedConnections field  clear only in the resetConnection method ？
if ChannelCachingConnectionProxy is not open,  clear this in allocatedConnections field on the destroy method

**Comment by garyrussell** [comment]
Yes; I noticed that last week but decided it wasn't very important because the connection will be refreshed the next time it is used:

https://github.com/spring-projects/spring-amqp/blob/05472a536aa97740c1be9d3adfd272c8eaf5f10d/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java#L795-L802

Feel free to submit a PR if you feel strongly about it.
